python-lib/wordcloud_generator.py

```diff
@@ -0,0 +1,192 @@
+    Attributes:
+        df (pandas.DataFrame): Dataframe containing text data
+        tokenizer (MultilingualTokenizer): Tokenizer used for text processing
+        text_column (str): Name of the df column on which to compute wordclouds
+        language (str, optional): Language in which to tokenize data (for monolingual wordcloud), defaults to english.
+            Use "Detected language column" for multilingual tokenization
+        language_column (str, optional): Name of the language column
+        subchart_column (str, optional): Name of the subcharts column to compute wordclouds on, defaults to None
+        max_words (int, optional): Maximum number of words to display in wordcloud, defaults to 100
+    """
+
+    DEFAULT_MAX_WORDS = 100
+    DEFAULT_NUM_PROCESS = 2
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        tokenizer: MultilingualTokenizer,
+        text_column: AnyStr,
+        path: AnyStr,
+        language: AnyStr = "en",
+        language_column: AnyStr = None,
+        subchart_column: AnyStr = None,
+        max_words: int = DEFAULT_MAX_WORDS,
+    ):
+        # Initialization method for the MultilingualTokenizer class, with optional arguments etailed above
+
+        self.df = df
+        self.tokenizer = tokenizer
+        self.text_column = text_column
+        self.language = language
+        self.language_column = language_column
+        self.subchart_column = subchart_column
+        self.path = path
+        self.max_words = max_words
+
+    def _color_func(self, word, font_size, position, orientation, random_state=None, **kwargs):
+        # Return the color function used in the wordcloud
+        color_list = [
+            "hsl(205,71%,41%)",
+            "hsl(214,56%,80%)",
+            "hsl(28,100%,53%)",
+            "hsl(30,100%,74%)",
+            "hsl(120,57%,40%)",
+            "hsl(110,57%,71%)",
+        ]
+
+        return random.choice(color_list)
+
+    def _get_wordcloud(self, frequencies, scale=6.8):
+        # Return a wordcloud as a svg file
+        wordcloud = (
+            WordCloud(background_color="white", scale=scale, margin=4, max_words=self.max_words)
+            .generate_from_frequencies(frequencies)
+            .recolor(color_func=self._color_func, random_state=3)
+        )
+
+        return wordcloud
+
+    def _prepare_data(self):
+        start = time()
+        logging.info("Preparing data")
+        if self.subchart_column != None:
+            # Group data per language and subchart for tokenization
+            group_columns = [col for col in [self.language_column, self.subchart_column] if col != None]
+            self.df.dropna(subset=group_columns, inplace=True)
+            self.df_grouped = self.df.groupby(group_columns)
+        else:
+            # Simply format data similarly
+            self.df_grouped = [(self.language, self.df)]
+        logging.info("Data preparation: Done in {:.2f} seconds.".format(time() - start))
```

Comment:
I have a preference for Python 3.6 f-string formatting. Not a big deal, but it's more readable